backend/tester/src/test/scala/io/theorg/tester/OrgchartApiTestSuite.scala

```diff
@@ -0,0 +1,136 @@
+
+  def withServer(
+      test: SttpBackend[Task, ZioStreams with WebSockets] => ZIO[
+        TestEnvironment,
+        Throwable,
+        TestResult
+      ]
+  ): ZIO[TestEnvironment, Throwable, TestResult] = {
+    val testJob =
+      for {
+        backend <- HttpClientZioBackend.managed()
+        response <- Managed.fromEffect(test(backend))
+      } yield response
+
+    testJob.useNow
+  }
+
+  val orgChartSelection = OrgChart.nodes(
+    (OrgChartNode.id.map(
+      UUID.fromString
+    ) ~ OrgChartNode.title ~ OrgChartNode.parentId.map(
+      (_.map(UUID.fromString))
+    )).mapN(OrgChartNodeMapped(_, _, _))
+  )
+
+  def getNodes(level: Int = 0) = Query.loadOrgChart(level)(
+    orgChartSelection
+  )
+
+  def addNode(title: String, parentId: Option[UUID]) = {
+    Mutation.addNode(AddNodeInput(title, parentId.map(_.toString())))(
+      AddNodeResult.orgChart(orgChartSelection)
+    )
+  }
+
+  private val calibanServerSpec =
+    suite("Caliban server Spec")(
+      testM("Empty on first run") {
+        withServer { backend =>
+          val response = getNodes().toRequest(apiUrl).send(backend)
+          assertM(response.map(_.code.code))(equalTo(200))
+          assertM(response.map(_.body).absolve.map(_.flatten.map(_.size)))(
+            equalTo(Some(0))
+          )
+        }
+      },
+      testM("Concurrent updates don't break the requirements") {
+        withServer { backend =>
+          for {
+            stateRef <- Ref.make(Set(Option.empty[UUID]))
+            _ <- ZIO.foreachParN_(AddNodeConcurrency)(0 to MaxAddNodeRequests)(
+              _ => {
+                stateRef.get
+                  .map(state => {
+                    val n = util.Random.nextInt(state.size)
+                    state.iterator.drop(n).next
+                  })
+                  .flatMap(nextParent =>
+                    addNode("node title", nextParent)
+                      .toRequest(apiUrl)
+                      .send(backend)
+                      .map(_.body)
+                      .absolve
+                      .map(_.flatten.flatten)
+                      .map(
+                        _.map(_.map(_.parentId))
+                          .getOrElse(List(Option.empty[UUID]))
+                          .distinct
+                          .toSet
+                      )
+                      .flatMap(latestState => stateRef.update(_ => latestState))
+                      .ignore
+                  )
+
+              }
+            )
+            response <- getNodes().toRequest(apiUrl).send(backend)
+          } yield {
+            assert(response.code.code)(equalTo(200))
+
+            val responseNodes = response.body
+              .map(_.flatten)
+              .map(_.getOrElse(List.empty))
+              .getOrElse(List.empty)
+
+            val nodesOverThresholdByParent =
+              responseNodes
+                .groupBy(_.parentId)
+                .filter(_._2.size > MaxNodesPerLevel)
```

Comment:
Consider explicit pattern decomposition of the tuple to give the second component a meaningful name.